Mathlib/Analysis/Convex/Topology.lean

```diff
@@ -91,6 +91,49 @@ end stdSimplex
 /-! ### Topological vector spaces -/
 section TopologicalSpace
 
+variable [OrderedSemiring 𝕜] [AddCommGroup E] [Module 𝕜 E] [TopologicalSpace E]
+
+theorem convex_closed_sInter {S : Set (Set E)} (h : ∀ s ∈ S, Convex 𝕜 s ∧ IsClosed s) :
+    Convex 𝕜 (⋂₀ S) ∧ IsClosed (⋂₀ S) :=
+  ⟨fun _ hx => starConvex_sInter fun _ hs => (h _ hs).1 <| hx _ hs,
+    isClosed_sInter fun _ hs => (h _ hs).2⟩
+
+variable (𝕜)
+
+/-- The convex closed hull of a set `s` is the minimal convex closed set that includes `s`. -/
+@[simps! isClosed]
+def convexClosedHull : ClosureOperator (Set E) := .ofCompletePred (fun s => Convex 𝕜 s ∧ IsClosed s)
+  fun _ ↦ convex_closed_sInter
+
+variable {𝕜}
+
+theorem convex_convexClosedHull {s : Set E} :
+    Convex 𝕜 (convexClosedHull 𝕜 s) := ((convexClosedHull 𝕜).isClosed_closure s).1
+
+theorem isClosed_convexClosedHull {s : Set E} :
+    IsClosed (convexClosedHull 𝕜 s) := ((convexClosedHull 𝕜).isClosed_closure s).2
+
+theorem subset_convexClosedHull {s : Set E} : s ⊆ convexClosedHull 𝕜 s :=
+  (convexClosedHull 𝕜).le_closure s
+
+theorem convexClosedHull_min {s t : Set E} : s ⊆ t → Convex 𝕜 t ∧ IsClosed t →
+    convexClosedHull 𝕜 s ⊆ t := (convexClosedHull 𝕜).closure_min
+
+theorem convexHull_subseteq_convexClosedHull {s : Set E} :
+    (convexHull 𝕜) s ⊆ (convexClosedHull 𝕜) s :=
+  convexHull_min subset_convexClosedHull convex_convexClosedHull
+
+theorem convexClosedHull_eq_convexClosedHull_closure {s : Set E} :
+    convexClosedHull 𝕜 s = convexClosedHull 𝕜 (closure s) := subset_antisymm
+  (convexClosedHull_min (subset_trans subset_closure subset_convexClosedHull)
+    ⟨convex_convexClosedHull, isClosed_convexClosedHull⟩)
+  (convexClosedHull_min (closure_minimal subset_convexClosedHull isClosed_convexClosedHull)
+    ⟨convex_convexClosedHull, isClosed_convexClosedHull⟩)
```

Comment:
There should be a much simpler proof using the lemma I just mentioned along the lines of:
```
convexClosedHull 𝕜 s ⊆ convexClosedHull 𝕜 (closure s) ⊆ convexClosedHull 𝕜 (convexClosedHull 𝕜 s) = convexClosedHull 𝕜 s
```
The first is `ClosureOperator.monotone`, the second is the lemma I just mentioned, and the third is `ClosureOperator.idempotent`, which is a `simp` lemma. So, something like (untested) should work:
```suggestion
theorem convexClosedHull_eq_convexClosedHull_closure {s : Set E} :
    convexClosedHull 𝕜 s = convexClosedHull 𝕜 (closure s) :=
  subset_antisymm ((convexClosedHull 𝕜).monotone subset_closure) <| by
    simpa using ((convexClosedHull 𝕜).monotone closure_subset_convexClosedHull)
```

Reply:
Done, thanks.